File: 브루트포스/6603.py
import sys
input = sys.stdin.readline

def dfs(a, idx, lotto):
    if len(lotto) == 6:
        print(' '.join(map(str,lotto)))
        return
    if idx == len(a):
        return
    dfs(a, idx+1, lotto+[a[idx]])
    dfs(a, idx+1, lotto)

while True:
    n, *s = list(map(int,input().rstrip().split()))
    if n == 0:
        break
    dfs(s, 0, [])
    print()

File: 브루트포스/6603.java
import java.util.*;

public class Main {
    public static ArrayList<Integer> lotto = new ArrayList<>();
    public static void dfs(int[] s, int idx, int cnt){
        if(cnt == 6){
            for(int x : lotto){
                System.out.print(x+" ");
            }
            System.out.println();
            return;
        }
        int n = s.length;
        if(idx==n) return;
        lotto.add(s[idx]);
        dfs(s, idx+1,cnt+1);
        lotto.remove(lotto.size()-1);
        dfs(s, idx+1,cnt);
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while(true){
            int n = sc.nextInt();
            if(n==0) break;
            int[] a = new int[n];
            for(int i = 0; i<n;i++){
                a[i] = sc.nextInt();
            }
            dfs(a, 0,0);
            System.out.println();
        }
    }
}
